pkg/distro/image_config.go

```diff
@@ -138,6 +138,12 @@ type ImageConfig struct {
 	// IsoRootfsType defines what rootfs (squashfs, erofs,ext4)
 	// is used
 	IsoRootfsType *manifest.RootfsType `yaml:"iso_rootfs_type,omitempty"`
+
+	// VersionlockPackges uses dnf versionlock to lock a package to the version
+	// that is installed during image build, preventing it from being updated.
+	// This is only supported for distributions that use dnf4, because osbuild
+	// only has a stage for dnf4 version locking.
+	VersionlockPackages []string `yaml:"versionlock_packages,omitempty"`
```

Comment:
Nitpick: maybe we should include the `dnf4` in the option name, just a thought. it would make it easier also to determine the correct packages to add to the package set if the customization is used.

Reply:
My thinking here is that the option can be reused for dnf5 locking if and when we introduce it.  If that never happens though, you're right, calling it dnf4 would make it clearer.